test/integration/controller/jobs/jobset/jobset_controller_test.go

```diff
@@ -79,204 +80,300 @@ var _ = ginkgo.Describe("JobSet controller", ginkgo.Ordered, ginkgo.ContinueOnFa
+			util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, lookupKey, createdJobSet); err != nil {
+					return false
+				}
+				return !*createdJobSet.Spec.Suspend
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+			gomega.Expect(createdJobSet.Spec.ReplicatedJobs[0].Template.Spec.Template.Spec.NodeSelector).Should(gomega.HaveLen(1))
+			gomega.Expect(createdJobSet.Spec.ReplicatedJobs[0].Template.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
+			gomega.Expect(createdJobSet.Spec.ReplicatedJobs[1].Template.Spec.Template.Spec.NodeSelector).Should(gomega.HaveLen(1))
+			gomega.Expect(createdJobSet.Spec.ReplicatedJobs[1].Template.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(spotFlavor.Name))
+
+			ginkgo.By("checking the workload is finished when JobSet is completed")
+			createdJobSet.Status.Conditions = append(createdJobSet.Status.Conditions,
+				metav1.Condition{
+					Type:               string(jobsetapi.JobSetCompleted),
+					Status:             metav1.ConditionStatus(corev1.ConditionTrue),
+					Reason:             "AllJobsCompleted",
+					Message:            "jobset completed successfully",
+					LastTransitionTime: metav1.Now(),
+				})
+			gomega.Expect(k8sClient.Status().Update(ctx, createdJobSet)).Should(gomega.Succeed())
+			gomega.Eventually(func() bool {
+				err := k8sClient.Get(ctx, wlLookupKey, createdWorkload)
+				if err != nil {
+					return false
+				}
+				return apimeta.IsStatusConditionTrue(createdWorkload.Status.Conditions, kueue.WorkloadFinished)
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+		})
+
+		ginkgo.It("Should allow to create jobset with one replicated job replica count 0", func() {
+			mixJobSet := testingjobset.MakeJobSet("mix-jobset", ns.Name).ReplicatedJobs(
+				testingjobset.ReplicatedJobRequirements{
+					Name:        "replicated-job-1",
+					Replicas:    2,
+					Parallelism: 2,
+					Completions: 2,
+				}, testingjobset.ReplicatedJobRequirements{
+					Name:        "replicated-job-2-empty",
+					Replicas:    0,
+					Parallelism: 0,
+					Completions: 0,
+				},
+			).Queue(localQueue.Name).
+				Request("replicated-job-1", corev1.ResourceCPU, "250m").
+				Request("replicated-job-2-empty", corev1.ResourceCPU, "250m").
+				Obj()
+
+			jobSetLookupKey := types.NamespacedName{Name: mixJobSet.Name, Namespace: ns.Name}
+			ginkgo.By("create the jobset", func() {
+				gomega.Expect(k8sClient.Create(ctx, mixJobSet)).Should(gomega.Succeed())
+				gomega.Eventually(func() *bool {
+					createdJobSet1 := &jobsetapi.JobSet{}
+					gomega.Expect(k8sClient.Get(ctx, jobSetLookupKey, createdJobSet1)).Should(gomega.Succeed())
+					return createdJobSet1.Spec.Suspend
+				}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(true)))
```

Comment:
```suggestion
				gomega.Expect(k8sClient.Create(ctx, mixJobSet)).Should(gomega.Succeed())
				gomega.Eventually(func(g gomega.Gomega) *bool {
					createdJobSet1 := &jobsetapi.JobSet{}
					g.Expect(k8sClient.Get(ctx, jobSetLookupKey, createdJobSet1)).Should(gomega.Succeed())
					g.Expect(createdJobSet1.Spec.Suspend).ShouldNot(gomega.BeNil())
					g.Expect(createdJobSet1.Spec.Suspend).Should(gomega.BeTrue())
				}, util.Timeout, util.Interval).Should(gomega.Succeeded)
```

I would suggest passing the gomega instance via arg to increase debuggability.